jlm/rvsdg/lambda.hpp

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright 2018 Nico Reißmann <nico.reissmann@gmail.com>
+ * Copyright 2025 Helge Bahmann <hcb@chaoticmind.net>
+ * See COPYING for terms of redistribution.
+ */
+
+#ifndef JLM_RVSDG_LAMBDA_HPP
+#define JLM_RVSDG_LAMBDA_HPP
+
+#include <jlm/rvsdg/FunctionType.hpp>
+#include <jlm/rvsdg/graph.hpp>
+#include <jlm/rvsdg/structural-node.hpp>
+#include <jlm/rvsdg/substitution.hpp>
+#include <jlm/util/iterator_range.hpp>
+
+#include <optional>
+#include <utility>
+
+namespace jlm::rvsdg
+{
+
+/** \brief Lambda operation
+ *
+ * A lambda operation determines a lambda's name and \ref FunctionType "function type".
+ */
+class LambdaOperation : public rvsdg::StructuralOperation
+{
+public:
+  ~LambdaOperation() override;
+
+  LambdaOperation(std::shared_ptr<const FunctionType> type);
+
+  LambdaOperation(const LambdaOperation & other) = default;
+
+  LambdaOperation(LambdaOperation && other) noexcept = default;
+
+  LambdaOperation &
+  operator=(const LambdaOperation & other) = default;
+
+  LambdaOperation &
+  operator=(LambdaOperation && other) noexcept = default;
```

Comment:
Could be removed, no? Or is this not possible as we do not have an empty default constructor?